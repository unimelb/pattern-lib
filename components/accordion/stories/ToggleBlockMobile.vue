<template>
  <ToggleBlockMobile>
    <template slot="default">
      <h2>Arts</h2>
    </template>
    <template slot="hidden">
      <h4>As one of the worldâ€™s leading universities, we aspire to build on our distinguished traditions and create an innovative future.</h4>
      <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab adipisci alias, cumque, esse incidunt consequatur, accusantium odit blanditiis ipsam dolorem repellendus ut corporis earum, illum a maiores optio voluptate dicta.</p>
    </template>
  </ToggleBlockMobile>
</template>

<script>
import ToggleBlockMobile from 'components/toggle/ToggleBlockMobile.vue';

export default {
  components: { ToggleBlockMobile },
  readme: { html: true, source: false },
};
</script>
