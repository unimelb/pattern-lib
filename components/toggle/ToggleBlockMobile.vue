<template>
  <ToggleBlock
    ref="container"
    class="toggleblock--mobile">
    <template slot="default">
      <slot />
    </template>
    <template slot="hidden">
      <slot name="hidden" />
      <a
        class="toggleblock__footer"
        @click="toggle">Close panel</a>
    </template>
  </ToggleBlock>
</template>

<script>
import ToggleBlock from './ToggleBlock.vue';

export default {
  components: { ToggleBlock },
  methods: {
    toggle() {
      this.$refs.container.toggle();
    },
  },
};
</script>
