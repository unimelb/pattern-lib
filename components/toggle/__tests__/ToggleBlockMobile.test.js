import {
  shallow, mount,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import ToggleBlockMobile from '../ToggleBlockMobile.vue';

expect.extend(toHaveNoViolations);

describe('ToggleBlockMobile', () => {
  it('should match snapshot', () => {
    const result = shallow(ToggleBlockMobile).element;
    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    const html = mount(ToggleBlockMobile).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
