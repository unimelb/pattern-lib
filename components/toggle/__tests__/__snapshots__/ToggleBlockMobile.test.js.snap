// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleBlockMobile should match snapshot 1`] = `<!---->`;
